### PR TITLE
Libraries and training reactions related to benzene benzyne and C6H5 recombination

### PR DESCRIPTION
**Pull request by ajocher**
- benzene, benzyne and C6H5 recombination libraries added.
- Add new family group yne for benzyne and and remove forbidden group Diels_alder_addition for benzene.
- Adding training reactions from Comandini2011_BenzeneBenzyne to Diels_alder_addition.
- Adding training reactions from Tranter2010_C6H5recomb to R_Recombination, H_Abstraction and Disproportionation.
- Added library with updated HACA rates at 1 atm.
- Added H2_Loss family and training reactions from Kislov2005_Diels-Alder to H2_Loss and Diels_alder_addition.

### CODE REVIEW

**Comment by mliu49** [comment]
The database tests are failing with this error:
```
DatabaseError: Unexpected duplicate reaction C6H6 + H <=> C6H5 + H2 in kinetics library Frenklach2019_ProcCombInst. Reaction index 2 matches index 1.
```

File: input/kinetics/families/H2_Loss/groups.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+name = "H2_Loss/groups"
+shortDesc = u""
+longDesc = u"""
+
+"""
+
+template(reactants=["Di-H-ene"], products=["ene1","H2"], ownReverse=False)
+
+reverse = "none"
```
Comment (mliu49) [inline input/kinetics/families/H2_Loss/groups.py]: This should be the name for the reverse reaction, which can be whatever you want, e.g. "h_addition_ene" or something.

File: input/kinetics/families/H2_Loss/groups.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    label = "Di-H-ene",
+    group = 
+"""
+1 *1 C u0 p0 c0 {2,D} {6,S} {7,S}
+2 *2 C u0 p0 c0 {1,D} {3,S} {8,S}
+3 *3 C u0 p0 c0 {2,S} {4,S} {9,S} {10,S}
+4 *4 C u0 p0 c0 {3,S} {5,D} {11,S}
+5 *5 C u0 p0 c0 {4,D} {6,S} {12,S}
+6 *6 C u0 p0 c0 {1,S} {5,S} {13,S} {14,S}
+7 R u0 p0 c0 {1,S}
```
Comment (mliu49) [inline input/kinetics/families/H2_Loss/groups.py]: You don't need to include the R atoms, since leaving them out also means that they can be anything.

**Comment by mliu49** [comment]
@ajocher, I rebased this branch and cleaned up some of your commits. I dropped the whitespace changes because I don't think they're necessary. Don't worry about the Codacy report for RMG-database; it's primarily for RMG-Py.

I'm also adding a few commits to this branch, to allow benzene bonds to react in Disproportionation and Diels_alder_addition, to work with your new training reactions and planned changes to aromatic resonance in RMG-Py.

From the new commits I added, I discovered a bug in the kekulization module, which I patch in ReactionMechanismGenerator/RMG-Py#1500. That should be merged before this branch.

**Comment by mliu49** [comment]
Update:
I'm calculating rates using CBS-QB3 for acetylene addition to naphthalene for the Diels-Alder family. I was worried about over-prediction of Diels-Alder addition of acetylene to aromatic rings, since the only training reaction was in the reverse direction. I will add those as training reactions once the calculations finish, and then I think this PR should be complete and ready to be merged after testing.

**Comment by ajocher** [comment]
Will be replaced by updated PAHdevelop branch.

**Pull request closed without merging**